app/services/account_reset/delete_account.rb

```diff
@@ -0,0 +1,41 @@
+module AccountReset
+  class DeleteAccount
+    include ActiveModel::Model
+    include GrantedTokenValidator
+
+    def initialize(token)
+      @token = token
+    end
+
+    def call
+      @success = valid?
+
+      if success
+        destroy_user
+        notify_user_via_email_of_deletion
```

Comment:
Order ok here?

Reply:
Yes, it's the same order as before. We capture the user's email before the user is destroyed, so we can still send an email after destroying the user. I will update the order to avoid confusion.